os/binfmt/binfmt_execmodule.c

```diff
@@ -217,15 +217,21 @@ int exec_module(FAR struct binary_s *binp)
 	}
 #endif
 
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	uint32_t stack_alloc_size = binp->stacksize + CONFIG_MPU_STACK_GUARD_SIZE;
+#else
+	uint32_t stack_alloc_size = binp->stacksize;
+#endif
+
 #ifdef CONFIG_APP_BINARY_SEPARATION
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	ARCH_GET_RET_ADDRESS
-	stack = (FAR uint32_t *)mm_malloc(binp->uheap, binp->stacksize, retaddr);
+	stack = (FAR uint32_t *)mm_malloc(binp->uheap, stack_alloc_size, retaddr);
 #else
-	stack = (FAR uint32_t *)mm_malloc(binp->uheap, binp->stacksize);
+	stack = (FAR uint32_t *)mm_malloc(binp->uheap, stack_alloc_size);
 #endif
 #else
-	stack = (FAR uint32_t *)kumm_malloc(binp->stacksize);
+	stack = (FAR uint32_t *)kumm_malloc(stack_alloc_size);
```

Comment:
When CONFIG_MPU_STACK_OVERFLOW_PROTECTION is enabled, we should use memalign function.
How about use up_createstack instead of using this separate allocation and calling up_use_stack?